course.py

```diff
@@ -135,8 +135,7 @@ def search_teachers():
 @login_required
 @check_access_level(Role.ADMIN)
 def course_info(course_id):
-    dynamic_year = get_current_year()
-    current_year = int(request.args.get('current_year')) if request.args.get('current_year') else dynamic_year
+    current_year = int(request.args.get('current_year'))
```

Comment:
Please avoid putting required arguments in request args: put them in the route, there are automatically sanitized.